CONTRIBUTE.md

```diff
@@ -0,0 +1,49 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make/new feature you want to add via issue, or any other method with the owners of this repository. 
+
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a 
+   build.
+2. Update the CHANGELOG.md with details of changes to the interface, this includes new environment 
+   variables, exposed ports, useful file locations and container parameters.
+3. Increase the version numbers in any examples files and the README.md to the new version that this
+   Pull Request would represent.
+4. Your PR will be merged (if accepted) once it is reviewed by other developers.
+5. It is advised to create a PR if you are stuck (so others can help you) or when your feature is complete and tested. 
+
+## New plugin template
+
+Details about creating a new plugin are listed [here](https://github.com/geometry-zsh/geometry/blob/master/plugins/README.md). See below for a common template that can be followed while creating a new plugin.
+
+* Add your plugin to `geometry/plugins/` folder 
+* To create a new plugin, you will need a setup, check, and a render function, with the plugin name on them.
```

Comment:
There is also a structure that users should follow. The plugin must be in a `plugin.zsh` file inside a directory with the name of the plugin.

Users are also advised to include a README with the plugin behavior as well as instructions to use and any customization options.